### DISCUSSION TITLE
Non-silent installers

### DISCUSSION THREAD

**Discussion by bquistorff** [microsoft/winget-pkgs#3388] [General] [answered]
Some publishers do not provide an installer with a simple `Silent` or `SilentWithProgress` (unattended install) option. What is preferred in this case?
1. Not submit a manifest for this program.
2. Submit a manifest without a `Silent` flag. (I know these PRs get flagged, but I'm not sure if they would eventually be accepted.)
3. (If possible) The contributor could repackage the installer to include a `Silent` flag. In the manifest, the `URL` of the installer then wouldn't match `Homepage`.

**Comment by Witchilich** [comment, marked as answer]
1 or 3. If it cant be installed from command line then do not put it into the manifest.

**Comment by denelon** [comment]
Is this related to installers that require a configuration file for silent install, or just packages that don't have a silent option at all?

**Reply by bquistorff** [reply]
Packages that don't have a silent option at all.

**Comment by denelon** [comment]
We have discussed the possibility of adding a hint to manifests for packages that can't / won't install silently. At least we could add settings or warnings for users attempting to install these packages.

**Reply by bquistorff** [reply]
That would certainly be easier on contributors (though potentially making it harder to audit/check all the packages).

**Reply by mr4550359** [reply]
Hey, can you tell me how to use silentwithopprogress or silent for winget pls. I want to contribute ?

**Reply by denelon** [reply]
https://github.com/microsoft/winget-cli/issues/910 needs to be implemented before we will start to accept "non-silent" installers in the community repository.

Be sure to add  your 👍 to raise the priority for that feature.

Until then, if you want to build the manifest add the link to that issue, then we can add the "Blocking-Issue" label to the PR to keep it open, and try again once 910 has been implemented.

**Comment by denelon** [comment]
@mr4550359 take a look at manifests with "InstallerType" of "exe". You can also look at the v1.0 manifest schema to see the complete syntax. *Note*: not all fields are implemented in the client yet (like dependencies).
